### PR TITLE
The reset column filter functionality is broken when the datagrid is nested inside another component

### PR DESCRIPTION
**Pull request by Jakub-Fajkus**
The reset column filter functionality is broken, when the datagrid component is nested inside another component.

Just to be clear, it is the X link in the table heading (screen included)

reset_column

So, the proposed solution is as follows.
I've removed the JS code, that changes the URL of the link to delete the filter on a column. The comment said that the code is there because table header is not refreshed using snippets. So, I've added the snippet to the `thead` element. Next, the snippet is invalidated in the `\Ublaboo\DataGrid\DataGrid::reload()` method.

Basically, I wonder why it was originally done that way and not via the snippet invalidation.

I've tested that change for grid, which is directly attached to the presenter as well as grid inside another component and it seems to be working just fine. i don't think this could cause any issue other than slightly bigger network traffic.

### CODE REVIEW

**Comment by Jakub-Fajkus** [comment]
@paveljanda @f3l1x What do you think? I would love to see that in the 5.7.* versions since I consider that a bug and because there are still people (inluding me) who cannot yet update to nette 3.0.

**Comment by f3l1x** [comment]
What do you think @paveljanda?

**Comment by paveljanda** [comment]
All this code was added because older (I am not sure whether it still happens in newer versions) nette versions did not behave exactly as intended - speaking about snippets. To merge this PR, it would require a thorough testing across multiple nette versions and all possible scenarious with both this and without this PR. :/

**Comment by f3l1x** [comment]
Please rebase your PR if it's still relevant.

**Comment by f3l1x** [comment]
Closing here, feel free to reopen it.
